### ISSUE TITLE
Speed up with native sparse matrices

### ISSUE THREAD

**Issue by zsunberg** [JuliaPOMDP/FIB.jl#8]
In the discretevalueiteration solver, we saw a 1-2 order of magnitude speedup from using builtin sparse matrices: https://github.com/JuliaPOMDP/DiscreteValueIteration.jl/blob/master/src/sparse.jl

It might also be possible to do this with FIB

**Comment by dressel** [comment]
I agree. If anyone wants to take a hack at it, go for it. I'm a bit occupied with other things at the moment.
